app/Http/Controllers/Api/AuthController.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Http\Controllers\Api;
+
+use App\Http\Controllers\Controller;
+use App\Http\Requests\LoginRequest;
+use App\Http\Requests\RegisterRequest;
+use App\Models\User;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Hash;
+use Illuminate\Support\Str;
+use Laravel\Socialite\Facades\Socialite;
+use Symfony\Component\HttpFoundation\Response;
+
+class AuthController extends Controller
+{
+    public function store(RegisterRequest $request): JsonResponse
+    {
+        $user = User::create([
+            "name" => $request->input("name"),
+            "email" => $request->input("email"),
+            "password" => Hash::make($request->input("password")),
+        ]);
+
+        return response()->json([
+            "message" => __("User created."),
+        ]);
+    }
+
+    public function login(LoginRequest $request): JsonResponse
+    {
+        $remember = $request->boolean("remember", false);
+
+        if (Auth::attempt([
+            "email" => $request->email,
+            "password" => $request->password,
+        ], $remember)) {
+            $user = Auth::user();
+            $user_id = (string)Auth::id();
+            $token_abilities = [];
+
+            if ($user->isAdmin()) {
+                $token_abilities = ["HasAdminRole"];
+            }
```

Comment:
Maybe move that into separated function, something like generator because u are doing the same some lines below - DRY principle.